### ISSUE TITLE
Poor image RTSP and Record Video SD Card

### ISSUE THREAD

**Issue by bmwm69** [ghoost82/mijia-720p-hack#4]
I use your hack on my Cam , RTSP works pretty well but the video quality is quite bad compared to the original MiHome .

With your hack I have some  stuttering/ artifacts in live rtsp and recorded videos on SDCARD, without hack I have clean image :|

[https://image.ibb.co/fvEA6w/Capture.jpg](url)

**Comment by max5962** [comment]
Same issue here :/

**Comment by ghoost82** [comment]
With the current RTSP server the stream quality is not as good as with the cloud stream. I will try to figure out if it is possible to improve it.

**Comment by ghoost82** [comment]
Do you have enabled cloud streaming and RTSP? If so this will explain the bad stream and recording quality. The CPU is not powerful enough to handle both simultaneously

**Comment by bmwm69** [comment]
I only have RTSP enabled

**Comment by zhaoparrot** [comment]
At first I got the same issue .
after I disabled cloud steaming ,the RTSP stream quality becomes good.
but it seems I can't turn the IR LEDs off after turn off cloud streaming, I'll test it some more.

**Comment by bmwm69** [comment]
Yes that feature dont work :| Night mode only work with cloud streaming :|

**Comment by zhaoparrot** [comment]
I think I'll solve this by unplug the wire for IR LED later

**Comment by bmwm69** [comment]
Ok but if you unplug the wire IR LED will work during the day and night , right?

**Comment by bmwm69** [comment]
AH ok , but my problem  is exactly the opposite . **No IR LED** during the night or day!

In my case I need IR LED ! the camera are located in garage without any source of light

**Comment by ghoost82** [comment]
Currently I'm improve the firmware hack to support the night vison mode

**Comment by ghoost82** [comment]
Just uploaded a new release with prebuild binaries. This release has support for the auto night vision mode while the cloud service is disabled

**Comment by max5962** [comment]
@ghoost82 
Autonight seems buggy ^^
![screenshot_2018-06-29-23-00-52](https://user-images.githubusercontent.com/34659571/42114791-a0a7dcde-7bf0-11e8-9411-6a41ad6079b9.png)

**Comment by Djowie** [comment]
I've got  the same issue as the original poster has issued before. Lot of artifacts during the rtsp stream, everything is disabled except voor RTSP,HTTP and SSH. Does anyone know a solution to the annoying artifacts in the stream?
```
## Prefered NTP server
NTP_SERVER="pool.ntp.org"
## Start syslog (0/1)
ENABLE_SYSLOG=0
## Total Disconnect from Xioami servers, includes disable of streaming and OTA updater (0/1)
DISABLE_CLOUD=1
## Manage the camera via cloud but prevent online updates (0/1)
DISABLE_OTA=1
## Start telnet server at system boot (0/1)
ENABLE_TELNETD=0
## Start SSH server at system boot (0/1)
ENABLE_SSHD=1
## Start web server at system boot (0/1)
ENABLE_HTTPD=1
## Start FTP server at system boot (0/1)
ENABLE_FTPD=0
## Start samba server at system boot (0/1)
ENABLE_SAMBA=0
## Start local streaming server at system boot (0/1)
ENABLE_RTSP=1

```

**Comment by max5962** [comment]
@ghoost82 
Any news about  night vision ? 
To retrive night vision i have to : 
./S99auto_night_mode stop
Stopping Auto Night Mode
OK
Then : 
 ./night_mode.sh on ( personnal script to use shells functions in function.sh)
But the quality is really bad

**Comment by max5962** [comment]
@ghoost82 any news ?

**Comment by gianky69** [comment]
@max5962: which commands do you send to Mijia 360 in order to move it via Home Assistant? I'm looking for these...
Thanks

**Comment by ziomik** [comment]
No good streaming :( 
streaming: https://www.youtube.com/watch?v=jkQc2-Db6Bs

**Comment by dcsg** [comment]
@max5962 how did you setup the controls in home assistant? I tried to run the `sh functions.sh motor left` in the camera itself via ssh without success.
